sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -679,7 +679,25 @@ class SQLContext(@transient val sparkContext: SparkContext)
    * @since 1.3.0
    */
   def dropTempTable(tableName: String): Unit = {
-    cacheManager.tryUncacheQuery(table(tableName))
+    dropTempTable(tableName, ifExists = false)
+  }
+
+  /**
+   * Drops the temporary table with the given table name in the catalog. If the table has been
+   * cached/persisted before, it's also unpersisted.
+   *
+   * @param tableName the name of the table to be unregistered.
+   * @param ifExists whether ignore non-existent table
+   *
+   * @group basic
+   * @since 1.5.0
+   */
+  def dropTempTable(tableName: String, ifExists: Boolean): Unit = {
```

Comment:
But users can also just check whether a table exists programmatically. I'd prefer not to add this API since there are a lot of ways to work around it.